package.json

```diff
@@ -1,30 +1,53 @@
 {
   "name": "probot",
-  "version": "0.0.0-development",
-  "description": "a trainable robot that responds to activity on GitHub",
+  "version": "7.0.0-typescript.1",
+  "publishConfig": {
+    "tag": "next"
+  },
+  "description": "a trainable robot that responds to activity on GitHub.",
   "repository": "https://github.com/probot/probot",
   "main": "lib/index.js",
   "bin": {
     "probot": "./bin/probot.js"
   },
   "scripts": {
+    "build": "rimraf lib && tsc -p tsconfig.json",
     "start": "node ./bin/probot run",
-    "test": "jest --coverage && standard && npm run doc-lint",
+    "test": "jest --coverage --forceExit && standard && npm run doc-lint",
     "doc-lint": "standard docs/**/*.md",
     "doc": "jsdoc -c .jsdoc.json",
     "semantic-release": "semantic-release pre && npm publish && semantic-release post",
-    "postsemantic-release": "script/publish-docs"
+    "postsemantic-release": "script/publish-docs",
+    "prepublish": "npm run build"
```

Comment:
When I run this I get:

```
npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.
```

Reply:
What if `npm run build` was executed during `prepack` and `pretest` rather than `prepublish`? 

That way the TypeScript files would be compiled right before publishing to npm and right before tests ran.

https://docs.npmjs.com/misc/scripts has more on the various scripts options.

Reply:
I would just change it to `prepare`, I did that on atom-beautify.  They're the same thing as noted here:  https://docs.npmjs.com/misc/scripts#deprecation-note